UI/NpcList/NPCRow.gd

```diff
@@ -0,0 +1,54 @@
+extends PanelContainer
+
+onready var _npcNameLabel = $HBoxContainer/Name
+onready var _npcGenderLabel = $HBoxContainer/Gender
+onready var _npcPersonalityLabel = $HBoxContainer/Personality
+onready var _amountOfChildrenLabel = $HBoxContainer/Children
+onready var meetButton = $HBoxContainer/CenterContainer2/Meet
+onready var forgetButton = $HBoxContainer/CenterContainer/Forget
+onready var showNpc = $ShowNPC
+var _npcID: String
+
+
+func setName(name: String):
+	_npcNameLabel.text = name
+
+
+func setGender(gender: String):
+	_npcGenderLabel.text = gender
+
+
+func setPersonality(personality: String):
+	_npcPersonalityLabel.text = personality
+
+
+func setChildren(children):
+	_amountOfChildrenLabel.text = String(children)
+
+
+func setNpcID(ID: String):
+	if(_npcID == null):
+		push_error("Exception: Attempt to set null NPC in the NPC list")
+	else:
+		_npcID = ID
+
+
+func getNpcID():
+	if(_npcID != null):
+		return _npcID
+	else: 
+		push_error("Exception: NPCRow: NPC ID is missing")
+
+
+func _on_Forget_pressed():
+	if(_npcID == null):
+		push_error("Exception: Attempt to delete null NPC in the NPC list")
+	else:
+		var npcListObj = load("res://UI/NpcList/NpcList.gd").new()
+		npcListObj.forgetNPC(_npcID)
```

Comment:
this looks very strange and I think you're leaking memory x3. You can either move the forgetNPC function into this class or create a signal and subscribe to it inside the NpcList. Signals are like events basically, perfect for this. Here is an example https://github.com/Alexofp/BDCC/blob/main/UI/SkillsUI/PerkButton.gd (look for the signal and how its being used)

you basically create a signal in this class `signal onForgetButtonPressed(npcID)`
then you emit it in this function `emit_signal("onForgetButtonPressed", _npcID)`
And then inside the NpcList class you subscribe to it dynamically while creating the rows `var _ok = someNpcRow.connect("onForgetButtonPressed", self, "forgetNPC")`
should work ^^

Reply:
Yes I wanted to make it with signals originally but creating an object **only** when you actually want to delete character seemed like a better choice than connecting **every** row.